executor/aggregate.go

```diff
@@ -816,6 +850,106 @@ func (e *HashAggExec) getPartialResults(groupKey string) []aggfuncs.PartialResul
 	return partialResults
 }
 
+func (e *HashAggExec) initRuntimeStats() {
+	if e.runtimeStats != nil {
+		if e.stats == nil {
+			e.stats = &HashAggRuntimeStats{
+				PartialNum:     e.ctx.GetSessionVars().HashAggPartialConcurrency(),
+				FinalNum:       e.ctx.GetSessionVars().HashAggFinalConcurrency(),
+				PartialTaskNum: 0,
+				FinalTaskNum:   0,
+				PartialTask:    0,
+				FinalTask:      0,
+				PartialTime:    make([]time.Duration, 0, e.ctx.GetSessionVars().HashAggPartialConcurrency()),
+				FinalTime:      make([]time.Duration, 0, e.ctx.GetSessionVars().HashAggFinalConcurrency()),
+			}
+			e.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.RegisterStats(e.id, e.stats)
+		}
+	}
+}
+
+// HashAggRuntimeStats record the HashAggExec runtime stat
+type HashAggRuntimeStats struct {
+	PartialNum int
+	FinalNum   int
+
+	PartialTaskNum int64
+	FinalTaskNum   int64
+
+	PartialTask time.Duration
+	FinalTask   time.Duration
+
+	PartialTime []time.Duration
+	FinalTime   []time.Duration
+}
+
+func (e *HashAggRuntimeStats) String() string {
+	var result string
+	if e.PartialTask != 0 {
+		result += fmt.Sprintf("partial_worker:{used_time:%s, PartialConcurrency:%d, task_num:%d", e.PartialTask, e.PartialNum, e.PartialTaskNum)
```

Comment:
How about renaming `used_time` to `wall_time`?
`PartialConcurrency` -> `concurrency`.